src/azul/portal_service.py

```diff
@@ -172,12 +173,15 @@ def _crud(self, operation: Callable[[JSONs], Optional[JSONs]]) -> Optional[JSONs
                     except NoSuchObjectVersion:
                         # Wait for latest version to appear in S3.
                         continue
-                db = operation(db)
-                if db is None:
-                    # Operation is read-only; we're done.
+                new_db = operation(db)
+                if db == new_db:
+                    # Operation didn't modify DB; we're done.
                     break
-                else:
+                elif db and new_db is None:
                     self._write_db(db, version)
```

Comment:
I suspect this change seemed necessary because of the lingering `None` in `test_portal_service`. With those addressed, I believe this branch should be removed.  Either `new_db` or `db` being `None` is a violation of `_crud`'s type annotation, which specifies that the operation must return `JSONs`, so I don't think there's any reason to branch based on their truthiness. Also, note that this dangerously conflates `db is not None` and with `db != []`. Also, this seems to write the same DB we just read, which would cause the same kind of duplication this PR is meant to fix.

I'm pretty sure there should be only two branches: one where the DB wasn't changed (`db == new_db`) and we don't write a new version, and one where it was changed (including possibly to `[]`) and we do.

Unrelated to this PR, but I find it odd that a method called `_crud` (**C**reate, **R**ead, **U**pdate, _**D**elete_) has no mechanism by which to delete the database. Just an idle observation, I don't think you need to address that here.